cmd/sealer/cmd/join.go

```diff
@@ -57,12 +59,37 @@ join default cluster:
 			return err
 		}
 
-		//TODO mount image and copy to new hosts
-		installer, err := cluster_runtime.NewInstaller(infraDriver, cf)
+		imageEngine, err := imageengine.NewImageEngine(imagecommon.EngineGlobalConfigurations{})
+		if err != nil {
+			return err
+		}
+
+		distributor, err := imagedistributor.NewScpDistributor(imageEngine, infraDriver)
 		if err != nil {
 			return err
 		}
 
+		var (
+			clusterImageName = cluster.Spec.Image
+			hosts            = append(newMasters, newWorkers...)
+		)
+
+		// distribute rootfs
+		if err = distributor.Distribute(clusterImageName, hosts); err != nil {
+			return err
+		}
+
+		runtimeConfig := new(clusterruntime.RuntimeConfig)
+		if cf.GetPlugins() != nil {
```

Comment:
First of all, I think it is not allowed to re specify a new clusterfile when joining. You must directly use the clusterfile used in init cluster, so that you can ensure that there is only one global clusterfile, which is simpler.

Therefore, the clusterfile object here should directly obtain the existing clusterfile in the cluster (a copy of the clusterfile should be kept in a k8s configmap after the cluster init is completed)

Reply:
What I think here is that the clusterfile in the `sealer run` phase is stored locally, and then when execute `sealer join`, hosts args will be merged with the local file. That is to say, there is no need to specify the clusterfile file in the `sealer join` and `sealer delete` cmds.  what do you think ?